legacy/expressions/migrate_test.go

```diff
@@ -300,35 +344,40 @@ func TestLegacyTests(t *testing.T) {
 	require.NoError(t, err)
 
 	var tests []legacyTest
-	err = json.Unmarshal(legacyTestData, &tests)
+	d := json.NewDecoder(bytes.NewReader(legacyTestData))
+	d.UseNumber()
+	err = d.Decode(&tests)
 	require.NoError(t, err)
 
 	for _, tc := range tests {
-		_, err := expressions.MigrateTemplate(tc.Template, expressions.ExtraAsFunction)
-
-		assert.NoError(t, err)
+		migratedTemplate, err := expressions.MigrateTemplate(tc.Template, expressions.ExtraAsFunction)
 
 		defer func() {
 			if r := recover(); r != nil {
 				t.Errorf("panic migrating template '%s': %#v", tc.Template, r)
 			}
 		}()
 
-		// TODO evaluate the migrated template? Would need to remap the provided variables, i.e. flow.x -> run.results.x
-		//
-		//if err == nil {
-		//	tz, err := time.LoadLocation(tc.Context.Timezone)
-		//	require.NoError(t, err)
-		//
-		//	env := test.NewTestEnvironment(utils.DateFormatDayMonthYear, tz, tc.Context.Now)
-		//
-		//	output, err := excellent.EvaluateTemplateAsString(env, tc.Context.Variables, migratedTemplate, tc.URLEncode, legacy.ContextTopLevels)
-		//
-		//	if len(tc.Errors) > 0 {
-		//		assert.Error(t, err, "expected error for template '%s' (migrated from '%s')", migratedTemplate, tc.Template)
-		//	} else {
-		//		assert.Equal(t, tc.Output, output, "output mismatch for template '%s' (migrated from '%s')", migratedTemplate, tc.Template)
-		//	}
-		//}
+		if err != nil {
+			assert.Equal(t, tc.Output, migratedTemplate, "migrated template should match input on error")
+		} else {
+			// evaluate the migrated template
+			tz, err := time.LoadLocation(tc.Context.Timezone)
+			require.NoError(t, err)
+
+			env := test.NewTestEnvironment(utils.DateFormatDayMonthYear, tz, tc.Context.Now)
+
+			migratedVars := tc.Context.Variables.Migrate()
+			migratedVarsJSON, _ := json.Marshal(migratedVars)
+
+			_, err = excellent.EvaluateTemplateAsString(env, migratedVars, migratedTemplate, tc.URLEncode, runs.RunContextTopLevels)
+
+			if len(tc.Errors) > 0 {
+				assert.Error(t, err, "expecting error evaluating template '%s' (migrated from '%s') with context %s", migratedTemplate, tc.Template, migratedVarsJSON)
+			} else {
+				// TODO enable checking of output
```

Comment:
Enabling this produces a fair number of failures - some of which are clearly not important (`TRUE` vs `true`) but will continue looking through them to find ones that do matter (see https://github.com/nyaruka/goflow/issues/274)